include/odp/api/spec/proto_stats.h

```diff
@@ -104,6 +123,12 @@ odp_proto_stats_t odp_proto_stats_lookup(const char *name);
  *
  * Destroy a proto stats object already created.
  *
+ * - Before destroying proto stats object having ODP_PROTO_STATS_ID_TX_*
+ *   enabled, all PKTIO devices to which packets were Tx'ed earlier with
+ *   this proto stats object need to be stopped. This is to make sure that
+ *   all the packets in Tx are flushed and there are no inflight packets
+ *   referring to this object.
```

Comment:
I am nos sure if this is adequate. What about packets inside TM or IPsec? Stopping a pktio device does not make sure that there are no packets in flight inside TM or inline outbound IPsec.

Reply:
For TM/Outbound IPSec, isn't it equivalent to stopping the PKTIO while TM/Outbound IPSec is running ?

Reply:
I mean, platform should already be handling by dropping existing packets and/or reporting error on enqueuing.

Reply:
I am not sure how exactly odp_pktio_stop() and TM & IPsec are supposed to interact. For odp_pktio_stop() the spec says: "Packets already stored for transmit are not freed. A following odp_packet_start() call restarts packet receive and transmit.". So if the in-flight packets are not freed (but in IPsec and TM cases they actually might, in linux-gen), then calling odp_pktio_stop() does not flush the Tx as the proposed API text now says.

Reply:
Modified the API text to mandate odp_pktio_stop() before destroying. It can be implementation defined as they will have to make sure proto stats object destroy is safe to be called.